Description: Trust __LP64__ on GNUC.
 Use __LP64__ to detect 64 bit systems, rather than trying to infer that
 from processor type.  This broke __x86_64__ in 32-bit mode (aka x32).
Author: Adam Borowski <kilobyte@angband.pl>

--- iceweasel-34.0.orig/js/src/jstypes.h
+++ iceweasel-34.0/js/src/jstypes.h
@@ -141,8 +141,7 @@
 # endif
 #elif defined(__GNUC__)
 /* Additional GCC defines are when running on Solaris, AIX, and HPUX */
-# if defined(__x86_64__) || defined(__sparcv9) || \
-        defined(__64BIT__) || defined(__LP64__)
+# if defined(__sparcv9) || defined(__64BIT__) || defined(__LP64__)
 #  define JS_64BIT
 # endif
 #elif defined(__SUNPRO_C) || defined(__SUNPRO_CC) /* Sun Studio C/C++ */
